src/composables/info/feature-info.composable.ts

```diff
@@ -0,0 +1,413 @@
+
+    for (let i = layers.length - 1; i >= 0; i--) {
+      const metadata = layers[i].get('metadata')
+      if (metadata !== undefined && metadata !== null) {
+        if (
+          metadata['is_queryable'] &&
+          layers[i].getVisible() &&
+          layers[i].getOpacity() > 0
+        ) {
+          const queryableId = layers[i].get('queryable_id')
+          layersList.push(queryableId)
+          layerLabel[queryableId] = layers[i].get('label')
+        }
+      }
+    }
+
+    if (layersList.length > 0) {
+      const resolution = map.getView().getResolution()
+      let bigBuffer = 0
+      let smallBuffer = 0
+      if (resolution) {
+        bigBuffer = 20 * resolution
+        smallBuffer = 1 * resolution
+      }
+
+      const point = transform(
+        evt.coordinate,
+        map.getView().getProjection(),
+        PROJECTION_LUX
+      )
+      const big_box = [
+        [point[0] - bigBuffer, point[1] + bigBuffer],
+        [point[0] + bigBuffer, point[1] - bigBuffer],
+      ]
+      const small_box = [
+        [point[0] - smallBuffer, point[1] + smallBuffer],
+        [point[0] + smallBuffer, point[1] - smallBuffer],
+      ]
+
+      setLoading(true)
+      map.getViewport().style.cursor = 'wait'
+
+      const size = map.getSize() || [0, 0]
+      const extent = map.getView().calculateExtent(size)
+      const bbox = extent.join(',')
+      const params = {
+        layers: layersList.join(),
+        box1: big_box.join(),
+        box2: small_box.join(),
+        srs: PROJECTION_WEBMERCATOR,
+        zoom: map.getView().getZoom(),
+        .../*(!map.get('ol3dm').is3dEnabled() &&*/ {
+          BBOX: bbox,
+          WIDTH: size[0],
+          HEIGHT: size[1],
+          X: evt.pixel[0],
+          Y: evt.pixel[1],
+        },
+      }
+
+      const url = new URL(INFO_SERVICE_URL)
+      Object.keys(params).forEach(key =>
+        url.searchParams.append(key, (params as Record<string, any>)[key])
+      )
+
+      try {
+        const resp = await fetch(url.toString())
+        if (resp.ok) {
+          const data = await resp.json()
+          if (data.length > 0) {
+            showInfo(evt.originalEvent.shiftKey, data, layerLabel, true, false)
+          } else {
+            lastHighlightedFeatures.value = []
+            featureInfoLayerService.highlightFeatures(
+              lastHighlightedFeatures.value,
+              false
+            )
+            reset()
+          }
+        } else {
+          throw new Error('Network response was not ok')
+        }
+      } catch (error) {
+        reset()
+      }
+    }
+  }
+
+  function showInfo(
+    shiftKey: boolean,
+    data: FeatureInfoJSON[],
+    layerLabel: { [key: string]: string },
+    openInfoPanel: boolean,
+    fit: boolean
+  ) {
+    if (shiftKey) {
+      data.forEach(item => {
+        item['layerLabel'] = layerLabel[item.layer]
+        let found = false
+        for (let iLayer = 0; iLayer < responses.value.length; iLayer++) {
```

Comment:
you don't like array functions anymore? ^^ I guess it is a copy past from v3 ;)

Reply:
Sorry, what do you mean here?